test/cppsuite/test_harness/workload/workload_tracking.cpp

```diff
@@ -173,4 +189,39 @@ workload_tracking::save_schema_operation(
         testutil_die(EINVAL, error_message.c_str());
     }
 }
+
+int
+workload_tracking::save_operation(const tracking_operation &operation,
+  const uint64_t &collection_id, const char *key, const char *value, wt_timestamp_t ts,
+  scoped_cursor &op_track_cursor)
+{
+    WT_DECL_RET;
+    std::string error_message;
+
+    if (!_enabled)
+        return (0);
+
+    testutil_assert(op_track_cursor.get() != nullptr);
+
+    if (operation == tracking_operation::CREATE_COLLECTION ||
+      operation == tracking_operation::DELETE_COLLECTION) {
+        error_message =
+          "save_operation: invalid operation " + std::to_string(static_cast<int>(operation));
+        testutil_die(EINVAL, error_message.c_str());
+    } else {
+        populate_tracking_cursor(operation, collection_id, key, value, ts, op_track_cursor);
+        ret = op_track_cursor->insert(op_track_cursor.get());
+    }
+    return (ret);
+}
+
+int
```

Comment:
You probably want this function to return `void` as `set_key` and `set_value` return `void`.

Reply:
Yep, fixed in e508f6a612fa3032d303ee01638dfa4a55030a69.
I must be missing some compilation flags in my local config..

Reply:
You probably meant ee0d40bb4aec6f0ebee5f97b330156f4a1449f12